tests/func/test_update.py

```diff
@@ -72,3 +74,48 @@ def test_update_import_url(repo_dir, dvc_repo):
     assert os.path.exists(dst)
     assert os.path.isfile(dst)
     assert filecmp.cmp(src, dst, shallow=False)
+
+
+def test_update_rev(dvc_repo, erepo):
+    src = "version"
+    dst = src
+
+    stage = dvc_repo.imp(erepo.root_dir, src, dst, rev="branch")
+
+    # update data
+    repo = Repo(erepo.root_dir)
+
+    saved_dir = os.getcwd()
+    os.chdir(erepo.root_dir)
```

Comment:
using os.chdir is not safe, as in case of an exception, you'll leave test suite in a different state that it was before running this. You should at least wrap it into try&finally.

Reply:
Also, you could take a look at `tests/func/test_import.py` where we do similar manipulations. E.g.https://github.com/iterative/dvc/blob/0.80.0/tests/func/test_import.py#L95 . It is using our new fixtures, which would be nice too use too, but we do realise that you've used the older ones here to match other tests in this repo, so it doesn't matter that much :slightly_smiling_face:

Reply:
Thank you for the heads up, the new testing util seems legit!

done by 32494bb7cfda45e7705fbca8ad00cab167757a94